### ISSUE TITLE
Location tab scrolling in Indicator Report UI

### ISSUE THREAD

**Issue by n1207n** [unicef/etools-partner-reporting-portal#1071]
When the Indicator Report has a lot of locations, the location tab section grows longer. As an result, the user has to scroll the tab section longer especially in the narrow height of the container.

We could try to stack the tabs instead of a single horizontal container. The width of this tab container has to be responsive of its parent container, and this may dictate # of tabs to store per stacked row.

![Capture_sites_shown_on_the_template_Karongi_is_last_at_extreme_right.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/58d2f674dc6eec1a00ff72e7/125516c66e82c728ace21e0d46cb958a379791acb0fab657a14ecf872149307d160bad14648de617ab477611030e0af92e230e06a3a7de72c9ef7f1fd60d48a9be3c1b1b937555eeeddf4ad043e377d30b97aaaeca0e9fb26876a399855d3b16564e81fc0835a39f429b59f34be14457a1143064e470246055d64f2b78343a7fe1a99d448d9762a5b0112a53c9e2e6ff151ba97dd8c31b3c.PNG)

**Comment by chachra** [comment]
@AttilaTheHen can you suggest options looking at how others in material design are handling this?

**Comment by AttilaTheHen** [comment]
After doing a little research, it looks like we've got a few options. Here's a pretty handy mockup I found of the different possible designs:

![tabs-alternatives](https://user-images.githubusercontent.com/27440940/53970482-ab5d8c80-40af-11e9-9b00-a4916b722e5b.png)

So #1 is what we've got now - a horizontal scrolling tab bar. We could stack the tabs like Silin suggested (#2), or have the tabs be displayed vertically on the side (#3), or have a dropdown (#4). #3 will probably take up too much screen real estate, so I think either #2 or #4 could work.

Personally I think a dropdown here makes the most sense - we don't end up taking more vertical space, and by opening the dropdown, we can quickly glance at a list of all locations and see whether they've been completed or not. Thoughts?

**Comment by n1207n** [comment]
In my opinion, vertical scrolling is easier than horizontal one. Even if we stack location tabs, the UI will look bloated and the organization will be broken quick.

I think #3 is the most appropriate one with one condition: a fixed container size. If we don't impose a fixed container size, then having 100 locations for data entry will not be scalable, since the container for data entry UI will be resized by its children.

This also means Reporting entity container, a tab row which shows UNICEF and Cluster right before data entry table, needs to move under.

In other words, when you expand Indicator row in Report UI, we need 2 container UIs: Location scrolling container and data entry table container with reporting entity tabs.

**Comment by brandinalexander** [comment]
Agreed - I feel there is plenty of room to accommodate this horizontally as well.

**Comment by n1207n** [comment]
Ideally 3-column space for location container and 9-column space for data entry container.

**Comment by brandinalexander** [comment]
**Rough Mockup:**
<img width="1233" alt="prp_location tab option" src="https://user-images.githubusercontent.com/33430685/53981741-2ee3a180-40e1-11e9-821f-e01d300812e3.png">

**Comment by n1207n** [comment]
We need to move the reporting entity tab into data entry table together.

<img width="1233" alt="53981741-2ee3a180-40e1-11e9-821f-e01d300812e3" src="https://user-images.githubusercontent.com/3138217/53981915-7a3d8600-40c8-11e9-8891-e13a0e1c97ce.png">

**Comment by AttilaTheHen** [comment]
Okay, #3 it is then!

**Comment by kasialanecka** [comment]
![Screenshot 2019-03-20 at 16.24.32.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/591488cfffff25001aeb9df0/125516c66e82c728ace21e0d46db978826878dba87e6ab03f60da1cd6416743e7b54e32a64c5b27ef1102f4f5c5d5cb053181d1ba6acda6fb4bc2e75820f5de5d2385f77c03442adbd821fb603bf278d4887b0ae965e82b96371b8d6d6056b5c17129bfe553fa19c4ecb58ff4ee5445cad1b3171ab.png)

@n1207n the idea for UI

**Comment by kasialanecka** [comment]
![Screenshot 2019-03-21 at 18.02.56.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/591488cfffff25001aeb9df0/125516c66e82c728ace21e0d46db978826878dba87e6ab03f60da1cd6416743e7b55e32a64c5b270f112294f5a595cb053181d1ba6acda6fb4bc2e75820f5de5d2385f77c03442adbd821fb603bf278d4887b0ae965e82b96371b8d6d6056b5c17129bfe553fa19c4ecb59fe4ce94d56a5113371ab.png)

@n1207n updated
